src/atomWithLocation.ts

```diff
@@ -50,13 +50,15 @@ type Options<T> = {
 type RequiredOptions<T> = Omit<Options<T>, 'getLocation' | 'applyLocation'> &
   Required<Pick<Options<T>, 'getLocation' | 'applyLocation'>>;
 
+type Option = { replace: boolean };
+
 export function atomWithLocation(
   options?: Options<Location>,
-): PrimitiveAtom<Location>;
+): WritableAtom<Location, [SetStateAction<Location>, Option?], void>;
```

Comment:
```suggestion
): WritableAtom<Location, [SetStateAction<Location>, AtomOptions?], void>;
```

Reply:
Done in https://github.com/jotaijs/jotai-location/pull/28/commits/df353647ce1470ee858b71a1e07320efe33759d4